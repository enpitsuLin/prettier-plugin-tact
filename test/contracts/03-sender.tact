import "@stdlib/deploy";

contract MessageSender with Deployable {

    deployer: Address;
    lastSender: Address;
 
    init() {
        self.deployer = sender(); // sender() of init is who deployed the contract
        self.lastSender = newAddress(0, 0); // zero address
    }

    receive("who") {
        if (sender() == self.deployer) {
            dump("deployer");
        } else {
            dump("not deployer!");
        }
    }
 
    receive("hello") {
        if (sender() != self.lastSender) {
            self.lastSender = sender();
            dump("hello new sender!");
        }
    }
}