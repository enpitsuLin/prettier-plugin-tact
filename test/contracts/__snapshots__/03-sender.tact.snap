import "@stdlib/deploy";

contract MessageSender with Deployable {
    deployer: Address;lastSender: Address;

    receive("who") {
        if (sender() == self.deployer) {
            dump("deployer");
        } else  {
            dump("not deployer!");
        }
    }

    receive("hello") {
        if (sender() != self.lastSender) {
            self.lastSender = ;
            dump("hello new sender!");
        }
    }
}